tests/customresources_test.go

```diff
@@ -0,0 +1,101 @@
+	t.Cleanup(cancel)
+
+	kube, err := internal.NewClient()
+	if err != nil {
+		t.Fatalf("Create client: %v", err)
+	}
+
+	t.Log("Testing that all core Crossplane CRDs exist and are well formed.")
+
+	wantCrds := []wantCRD{
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "configurations.pkg.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "configurationrevisions.pkg.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "providers.pkg.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "providerrevisions.pkg.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "functions.pkg.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "functionrevisions.pkg.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "deploymentruntimeconfigs.pkg.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "imageconfigs.pkg.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "compositeresourcedefinitions.apiextensions.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "compositions.apiextensions.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "compositionrevisions.apiextensions.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "environmentconfigs.apiextensions.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "managedresourcedefinitions.apiextensions.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "managedresourceactivationpolicies.apiextensions.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "usages.protection.crossplane.io"}}, namespaced: true},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "clusterusages.protection.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "usages.apiextensions.crossplane.io"}}},
+
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "operations.ops.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "cronoperations.ops.crossplane.io"}}},
+		{crd: kextv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: "watchoperations.ops.crossplane.io"}}},
+	}
+
+	for _, w := range wantCrds {
+		crd := w.crd
+		if err := kube.Get(ctx, types.NamespacedName{Name: crd.GetName()}, &crd); err != nil {
+			t.Errorf("Cannot get CRD %q: %v", crd.GetName(), err)
+			continue
+		}
+
+		// TODO(negz): Just cmp.Diff the entire CRD spec?
```

Comment:
This particular test does seem a little low value without actually doing this. 🤔 

The CRD could be completely different schematically, as long as it was e.g. cluster scoped and in the right categories.

Reply:
That's a fair point - I do like though that it is at least verifying the entire expected API surface area. This test will fail if a distribution doesn't have all the required APIs, which has some value in testing.